src/main/java/org/jenkinsci/plugins/emotional_jenkins/EmotionalJenkinsPublisher.java

```diff
@@ -35,50 +43,41 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         }
 */
 
-        build.getActions().add(new EmotionalJenkinsAction(build.getResult()));
-        return true;
-    }
-
-    public Action getProjectAction(Project project) {
-        return new EmotionalJenkinsAction();
+        build.addAction(new EmotionalJenkinsAction(build.getResult()));
     }
 
     public BuildStepMonitor getRequiredMonitorService() {
-        return BuildStepMonitor.BUILD;
+        return BuildStepMonitor.NONE;
     }
 
-    @Override
-    public BuildStepDescriptor<Publisher> getDescriptor() {
-        return DESCRIPTOR;
+    @Extension public static final class LastBuildActionFactory extends TransientActionFactory<Job> {
```

Comment:
Code style mess, this file contains normal styling with annotation on new line, please follow it.